File: ios/ReduxTask/DeviceCheck.m
//
//  DeviceCheck.m
//  ReduxTask
//
//  Created by Heli Shah on 21/09/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
@interface RCT_EXTERN_MODULE(DeviceCheck, NSObject)

RCT_EXTERN_METHOD(checkIsEmulator)
RCT_EXTERN_METHOD(getStatusDevice: (RCTResponseSenderBlock)callback)

@end

File: ios/DeviceCheck.swift
//
//  Bulb.swift
//  ReduxTask
//
//  Created by Heli Shah on 20/09/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation

@objc(DeviceCheck)
class DeviceCheck: NSObject {
  
@objc
  static var isDevice = false

  @objc
  func checkIsEmulator() {
   
    if Platform.isSimulator {
      DeviceCheck.isDevice = true
 print("Is simulator")
      
    } else {
        print("Is device")
      DeviceCheck.isDevice = false

    }
  }

  @objc
  func getStatusDevice(_ callback: RCTResponseSenderBlock) {
    callback([NSNull(), DeviceCheck.isDevice])
  }
  
@objc
static func requiresMainQueueSetup() -> Bool {
return true
}
}
struct Platform {
    static var isSimulator: Bool {
        #if targetEnvironment(simulator)
            // We're on the simulator
            return true
        #else
            // We're on a device
             return false
        #endif
    }
}
